superduperdb/backends/mongodb/query.py

```diff
@@ -733,7 +733,9 @@ def model_update(
         document_embedded = kwargs.get('document_embedded', True)
 
         if key.startswith('_outputs'):
-            key = key.split('.')[1]
+            # key format is _outputs.{field}.{model}.{version}
+            # extract new key = model
+            key = key.split('.')[2]
```

Comment:
At present, our design will cause ﻿key to be confusing when multiple listeners are connected in series. I have a new solution that may be able to solve it. I will add it to the discussion after 0.2 is released.

Reply:
Isn't this incorrect? This will cause the output to be saved in `_outputs.<model>....`. This could be ambiguous in the case in which the `<model>` is applied to multiple keys.

Reply:
Maybe we should make `key = _outputs:{field}:{model}:{version}` or something like that.

Reply:
Another thought - we should re-introduce optional specifiable `listener.identifier`. Then
we should store the outputs in `_outputs: {<identifier>/version: ...}` (not in `{'_outputs': {'key': {'model': {'version': ...}}}}`).

Reply:
Done. At present, I restored the original design and only hotfixed the bug when using multiple listeners and vector_index together.